Feature/LoginView implementation

caioalcn opened the pull request:
# Feature
- LoginView implementation;

# Checklist
- [x]  Implementar componente em um arquivo chamado LoginView.swift
- [x]  O componente deve mostrar o título "Login with your account” centralizado
- [x]  Botão principal deve mostrar o título "Login”
- [x]  Botão secundário deve mostrar o título "Forgot password”
- [x]  O componente deve ser mostrado na Tela de Login (LoginViewController)
- [x]  Tela de Login deve mostrar o título "Account”

Commit:
Feature:
- LoginView implementation;

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountView.swift:
```diff
@@ -0,0 +1,174 @@
+        label.numberOfLines = 0
+        label.textColor = .black
+        label.translatesAutoresizingMaskIntoConstraints = false
+        
```
Que tal remover este espacamento?

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountView.swift:
```diff
@@ -0,0 +1,174 @@
+        
+        NSLayoutConstraint.activate([
+            titleStackView.safeAreaLayoutGuide.topAnchor.constraint(equalTo: self.safeAreaLayoutGuide.topAnchor, constant: 15),
+            titleStackView.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 10),
```
Podemos alterar para left/right?

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountView.swift:
```diff
@@ -0,0 +1,174 @@
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setupLayout(){
```
Podemos quebrar o setupLayout em dois:

1. setupConstraints para as constraints
2. setupSubView para adicionar subviews e arranged subviews

Podemos deixar assim entao restruturando as propriedades que adicionam subviews?

Obrigado!

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountView.swift:
```diff
@@ -0,0 +1,174 @@
+        textField.autocorrectionType = .no
+        textField.delegate = self
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        
```
Se possivel remover as linhas vazias apenas por padronizacao

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountView.swift:
```diff
@@ -0,0 +1,174 @@
+    }
+}
+
+extension AccountView: UITextFieldDelegate{
```
espacamento depois de UITextFieldDelegate

UITextFieldDelegate {

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountView.swift:
```diff
@@ -0,0 +1,174 @@
+        return button
+    }()
+    
+    lazy var titleStackView: UIStackView = {
```
Private nas propriedades para serem usadas apenas na classe.

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountViewController.swift:
```diff
@@ -7,11 +7,50 @@
+    
+    private let viewModel = AccountViewModel()
+    
+    init() {
```
Podemos injetar a ViewModel no init

E usarmos o padrao Factory ou Coordinator para criar a cena da funcionalidade e injetar a camada responsavel.

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountViewModel.swift:
```diff
@@ -0,0 +1,30 @@
+
+import UIKit
+
+protocol AccountViewModelProtocol: AnyObject{
```
AnyObject {

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountViewModel.swift:
```diff
@@ -0,0 +1,30 @@
+
+protocol AccountViewModelProtocol: AnyObject{
+    var didLogin: (String) -> () { get set }
+    var didForgot: (String) -> () {get set }
```
{ get set }
espaços

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountViewModel.swift:
```diff
@@ -0,0 +1,30 @@
+final class AccountViewModel: AccountViewModelProtocol{
+    var didLogin: (String) -> () = {_ in }
+    var didForgot: (String) -> () = {_ in }
+
```
linhas, podemos remover aqui alguns?

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountViewModel.swift:
```diff
@@ -0,0 +1,30 @@
+    func doForgot()
+}
+
+final class AccountViewModel: AccountViewModelProtocol{
```
AccountViewModelProtocol {

Commit:
Fix:
- Removed spaces left;
- Break the setupLayout func in two, one to handle the adition of subviews and the other to set the constraint to the views;
- Change the leading/trailing constraint to left/right;
- Set private to variables that only the class uses;

Commit:
Fix:
- Remove spaces left;
- Use dependecy injection in AccountController to inject viewModel;

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/Account/AccountViewModel.swift:
```diff
@@ -7,18 +7,16 @@
     var didLogin: (String) -> () = {_ in }
     var didForgot: (String) -> () = {_ in }
-
     
```
remover linha

Commit:
Fix:
- Remove optional chaining from ViewModel in AccountViewController;
